doc/site/docs/01-getting-started/03-shelf-migration.md

```diff
@@ -0,0 +1,368 @@
+# Migration from Shelf
+
+This guide helps you migrate from [Shelf](https://github.com/dart-lang/shelf) to Relic. While the core concepts remain similar (handlers, middleware, requests, and responses), Relic introduces improvements in type safety and developer experience that require some changes to your code.
+
+### Why migrate?
+
+Relic was born out of the needs of [Serverpod](https://serverpod.dev) for a more modern web server foundation with stronger type safety. Shelf has been an excellent foundation for Dart web servers, but certain architectural decisions made years ago limit its ability to take advantage of modern Dart features.
+
+## Migration overview
+
+Use this quick plan to get your app running on Relic. The detailed sections below show code for each step.
+
+1) ✅ Update dependencies: Remove  `shelf`, `shelf_router`, `shelf_web_socket`. Add `relic`. If you serve over `dart:io`, use `relic/io_adapter` to start the server.
+
+2) ✅ Bootstrap the server: Replace `shelf_io.serve()` with `RelicApp().serve()` if using the io adapter, or integrate RelicApp into your hosting environment as needed.
+
+3) ✅ Update handler signatures: Change from `Response handler(Request request)` to use either `respondWith` or `ResponseContext handler(NewContext ctx)`.
+
+4) ✅ Switch to Relic routing: Replace Router from shelf_router with `RelicApp().get/post/put/delete`. Replace `<id>` path params with `:id` and read them via `ctx.pathParameters[#id]`.
+
+5) ✅ Create responses with Body: Replace `Response.ok('text')` with `Response.ok(body:...)`. Let Relic manage content-length and content-type through Body.
+
+6) ✅ Replace header access: Replace string lookups like `request.headers['cookie']` with typed accessors such as `request.headers.cookie`.
+
+7) ✅ Replace middleware and scoping: Replace `Pipeline().addMiddleware(...)` with `router.use(...)` and attach handlers under that path.
+
+8) ✅ Replace request.context usage: Replace `request.change(...)` and manual casts with `ContextProperty<T>().set/get` on the context.
+
+9) ✅ Update WebSockets: Replace `webSocketHandler` and use `RelicWebSocket` for events and sending.
+
+10) ✅ Satisfy the state machine: Handlers must return a handled context.
+
+## Detailed migration steps
+
+### 1. Update dependencies
+
+In your `pubspec.yaml`, remove Shelf packages and add Relic:
+
+Before:
+
+```yaml
+dependencies:
+  shelf: <shelf_version>
+  shelf_router: <shelf_router_version>
+  shelf_web_socket: <shelf_web_socket_version>
+```
+
+After:
+
+```yaml
+dependencies:
+  relic: <latest_version>
+```
+
+If you serve over `dart:io`, you'll use the `relic/io_adapter.dart` import to start the server.
+
+### 2. Bootstrap the server
+
+Before (Shelf):
+
+```dart
+import 'package:shelf/shelf_io.dart' as shelf_io;
+
+void main() async {
+  await shelf_io.serve(handler, 'localhost', 8080);
+}
```

Comment:
In shelf I suppose it is common to  use the `shelf_router` package, which will make the two look more similar.

Reply:
okay